src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-list/smb-cluster-list.component.ts

```diff
@@ -118,12 +118,21 @@ export class SmbClusterListComponent extends ListWithDetails implements OnInit {
       itemNames: [cluster_id],
       actionDescription: $localize`remove`,
       submitActionObservable: () =>
-        this.taskWrapper.wrapTaskAroundCall({
-          task: new FinishedTask('smb/cluster/remove', {
-            cluster_id: cluster_id
-          }),
-          call: this.smbService.removeCluster(cluster_id)
-        })
+        this.taskWrapper
+          .wrapTaskAroundCall({
+            task: new FinishedTask('smb/cluster/remove', {
+              cluster_id: cluster_id
+            }),
+            call: this.smbService.removeCluster(cluster_id)
+          })
+          .subscribe({
+            error: () => {
+              this.modalService.dismissAll();
```

Comment:
sill question, maybe related to UX, do we need to close the modal if the action is failed? we could just leave the modal there and force the user to close it so that he is a bit more aware about the fact that the actions is failed just in case he missed the notification

Reply:
Interesting point @nizamial09.
When I tried it, it did not make much sense to me leaving the modal open since the user won't be able to perform any other action than to close it, so I just saved the user one click by closing it automatically, but also and most important in my opinion is that when the modal is closed is easier to see and read notifications, otherwise those would "behind" the modal.

I think this way is easier to relate the modal closing and the notification message with the failure to perform the delete

Does that make sense, what do you think @nizamial09 ?

Reply:
Yup, makes sense. if the notification is behind the modal, then its definitely harder to read. Then one extra thing we should do is to make sure the notification is persisted in the notification sidebar (I saw sometimes the toast are just lost) and also rather than saying `Failure to execute SMB` can we have our own custom heading that is created in the UI? Something that states the right action the user was doing?

Reply:
just saw that the message is coming from the task-message service. I guess those messages are not persisted for some reason within the panel. don't know if its intentional or not. But probably that's a more generic bug that we need to look at.

But the current message looks good to me, thank you @Pegonzal